src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -156,64 +143,94 @@ object TezosDatabaseOperations {
     }
 
   /**
-    * Generates database rows from a list of averageFees calculated for some operation kinds.
-    * @param maybeFees  List of possible new fees that may have been calculated for different
-    *                   operation kinds.
-    * @return           Database rows
+    *
+    * @param fees      List of average fees for different operation kinds.
+    * @return          Database action possibly containing the number of rows written (if available from the underlying driver)
     */
-  def feesToDatabaseRows(maybeFees: List[Option[AverageFees]]): List[Tables.FeesRow] = {
-    maybeFees
-      .filter(_.isDefined)
-      .map{ someFee =>
-        val fee = someFee.get
-        Tables.FeesRow(
-          low = fee.low,
-          medium = fee.medium,
-          high = fee.high,
-          timestamp = fee.timestamp,
-          kind = fee.kind
-        )
-      }
-  }
-
+  def writeFeesIO(fees: List[AverageFees]): DBIO[Option[Int]] =
```

Comment:
Why convert feesToDatabaseRows to this RowConversion function you've defined later?

Reply:
I'm doing this because `RowConversion` is more simple in its responsibility, and can be reused as a composable piece of code in other computations.

Instead the `feesToDatabaseRows` was a very specialized operation that would have less chance for reuse, and can be expressed as a composition of other more basic operations anyway.

I've put it into a `RowConversion` to make the scope, and thus the responsibility of further similar converters added later on, clearer.

As a last point, you might've noticed that I split the responsibility to filter out possibly missing fees (`maybeFees`) from the actual conversion. That's because the fact that the input fees may not be there is not intrinsically related to writing fees to the database. I opted to shift this responsibility to the caller, which is in charge of putting things together by
 1. retrieving the fees
 2. removing those that are missing
 3. requesting to write the rest to db

Reply:
Thanks for the detailed response! And your last paragraph makes sense! I'll keep this design decision in mind